### PR TITLE
Convert to Ember CLI plugin

### PR DESCRIPTION
**Pull request by Robdel12**
## ~~This is still a work in progress~~ I think it's good!
- [x] TODO: Successfully `npm link` in a project and test the generator 
- [x] TODO: Clean up code a little
- [x] TODO: Update package json
- [x] TODO: Get JSON editor to work
- [x] TODO: Build so gh pages will work still.
- [x] TODO: Add GH-Pages builder to project: https://github.com/poetic/ember-cli-github-pages
- [ ] TODO: Update to use ember-jsoneditor CLI addon (@Glavin001 it doesn't look like it's published as an ember addon?)

This takes care of #14
## Notes

@Glavin001 I had to edit ember-jsoneditor for it to work. 

[Here I had to](https://github.com/Glavin001/ember-jsoneditor/blob/master/build/lib.js#L52) change the code to:
```javascript
var editor = new JSONEditor(container, options, json);
```

### CODE REVIEW

File: addon/components/c3-chart.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    Data Observer
+  */
+  dataDidChange: function() {
+    var self = this;
+    var chart = self.get('chart');
+    chart.load(self.get('data'));
+  }.observes('data').on('didInsertElement')
+});
+
+// Ember.Handlebars.helper('c3-chart', Ember.C3.ChartComponent);
```
Comment (Robdel12) [inline addon/components/c3-chart.js]: - [ ] TODO: Remove dead code

**Comment by Glavin001** [comment]
Is ember-cli-github-pages still a TODO?

**Comment by Robdel12** [comment]
Ah yes! That way the example on the gh-pages branch will be up to date. I can do this after the Super Bowl tonight.

**Comment by MiguelMadero** [comment]
It would be nice to ckage the c3 lib as pure json for those non-embercli consumers.

**Comment by Robdel12** [comment]
@MiguelMadero would keeping the current state of the project on a separate branch work?

**Comment by MiguelMadero** [comment]
@Robdel12 I guess that's a simple option, but then those would be two different versions to keep in sync/publish. For one of our internal libraries, we started shipping a pre-built lib that doesn't depend on EmberCLI. It's not too hard, and more ember-c3 it should be straightforward. 

[Packaging Ember CLI Add-ons](http://discuss.emberjs.com/t/loading-ember-cli-addons-dynamically-on-application-start/6909/5?u=miguelmadero)

**Comment by Robdel12** [comment]
@Glavin001 did we want to revisit/revive this?

**Comment by Glavin001** [comment]
Oh yeah, @Robdel12. If I recall when I last tried this pull request I was unable to get the gh-pages demo to work. I am not currently using ember-c3 with any of my current projects so I do not have any personal uses to test it further and it's hard to find time to contribute back to this particular project. 
A simply demo in the gh-pages branch would be sufficient to make sure this works and does not break anything. Essentially take gh-pages and make it using the Ember-CLI rewrite you developed.
This weeks I'm going to be working on other things but I should be able to check if this PR is in working order at the end of the week and we can get this all merged up. Thanks!

**Comment by Robdel12** [comment]
So I updated it to the latest ember and had to remove the JSONEditor since that's now borked too. And there's a bash script used to deploy to Gh-pages.. Which I don't like but It.. works. Meh. 

http://robdel12.github.io/ember-c3/

I'll keep dorkin with it.

**Comment by Glavin001** [comment]
Great work!!

What was the problem with JSONEditor again?

> I'll keep dorkin with it.

Let me know once you are happy with your PR. I will be busy until at least Sunday this week, so I may get a chance to check it over then if you're satisfied with your work / implementation by then.

Keep up the great work!

**Comment by wvteijlingen** [comment]
:+1: It looks like this removes the deprecated use of `Ember.Handlebars.helper`. Can this be merged into a release, maybe just with the JSON editor removed from the demo page?

**Comment by Glavin001** [comment]
I now have a new project that uses both Ember-CLI and C3, and so I will be revisiting this package soon -- hopefully this upcoming week. I will also clean up JSON editor and make this all work as expected, removing deprecated APIs, etc.

Thank you everyone for your patience!

**Comment by Glavin001** [comment]
I have completed [`ember-jsoneditor`](https://github.com/Glavin001/ember-jsoneditor). Now that I know how to convert to a Ember-CLI and the `ember-jsoneditor` component is ready, I can finish this work off with `ember-c3`.

**Pull request merged**